index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
> Just saw #350. I'll wait on any changes until that discussion resolves.

I don't think `tagBy` invalidates `ifElse`. I think we should add `tagBy` but keep `ifElse`. If we're keeping `ifElse` we should add `when` and `unless` with type `(a -> Boolean) -> (a -> a) -> a -> a`.